Bugfixing small issues related to title links

alexmylonas opened the pull request:
## Description

Small PR fixing multiple of small bugs: 

- [Log Analysis Overview: Alert title clickable area extends across the entire page](https://github.com/panther-labs/panther/issues/1829)
- [Removed link from title on SourceCard on panther created log sources](https://github.com/panther-labs/panther/issues/1763)

- [Log Analysis Rules - UX - Allow text entry in Log Type box](https://github.com/panther-labs/panther/issues/1894)
- [Log Analysis Rules - Timestamp set incorrectly](https://github.com/panther-labs/panther/issues/1892
)

Closes #1829, closes #1763, closes #1894, closes #1892 

## Tests 

- Locally

3nvi commented on web/src/components/cards/AlertCard/AlertCard.tsx:
```diff
@@ -44,14 +44,16 @@ const AlertCard: React.FC<AlertCardProps> = ({ alert, hideRuleButton = false })
-          <GenericItemCard.Heading>{alert.title}</GenericItemCard.Heading>
-        </Link>
+        <GenericItemCard.ValuesGroup>
+          <Link
```
This fixes the issue at the `AlertCard` level, but this is a `GenericItemCard` issue. I'd suggest to simply add `truncated` prop on the `GenericItemCardHeading` component.

While there, you could also remove `cursor: pointer` from all Links that are wrapping a `GenericItemCard.Heading` component cause I believe it's not needed

alexmylonas replied:
Removed `cursor: pointer`. 

Revert my change and changed every instance of `GenericItemCard.Heading`:  
```
        <Link
          as={RRLink}
          aria-label="Link to Alert"
          to={urls.logAnalysis.alerts.details(alert.alertId)}
          cursor="pointer"
        >
          <GenericItemCard.Heading>{alert.title}</GenericItemCard.Heading>
        </Link>
```
To: 
```
          <GenericItemCard.Heading>
            <Link
              as={RRLink}
              aria-label="Link to Rule"
              to={urls.logAnalysis.rules.details(rule.id)}
            >
              {rule.displayName || rule.id}
            </Link>
          </GenericItemCard.Heading>
```

alexmylonas commented:
Addressed feedback, check above!

Also added fixes for:
- [Log Analysis Rules - UX - Allow text entry in Log Type box](https://github.com/panther-labs/panther/issues/1894)
- [Log Analysis Rules - Timestamp set incorrectly](https://github.com/panther-labs/panther/issues/1892
)

3nvi commented on web/src/pages/ListDestinations/DestinationCards/DestinationCard.tsx:
```diff
@@ -37,13 +37,15 @@ const DestinationCard: React.FC<DestinationCardProps> = ({ destination, logo, ch
+          <Link
+            as={RRLink}
+            to={urls.settings.destinations.edit(destination.outputId)}
+            cursor="pointer"
```
```suggestion

```

alexmylonas replied:
Removed!

3nvi commented on web/src/components/cards/RuleCard/RuleCard.tsx:
```diff
@@ -66,7 +67,10 @@ const RuleCard: React.FC<RuleCardProps> = ({ rule }) => {
             <Flex ml="auto" mr={0} align="flex-end" spacing={4}>
               <GenericItemCard.Value label="Threshold" value={formatNumber(rule.threshold)} />
-              <GenericItemCard.Value label="Time Created" value={formatDatetime(rule.createdAt)} />
+              <GenericItemCard.Value
```
why are we changing it to `rule. lastModified`?

alexmylonas replied:
According to this [issue](https://github.com/panther-labs/panther/issues/1892): 

> Expected behavior
> The rules overview page correctly displays the Last Modified time.

3nvi replied:
ah gotcha. Missed that.

3nvi commented on web/src/pages/ListLogSources/LogSourceCards/LogSourceCard.tsx:
```diff
@@ -67,13 +67,17 @@ const LogSourceCard: React.FC<LogSourceCardProps> = ({ source, children, logo })
+          </GenericItemCard.Heading>
+        ) : (
+          <GenericItemCard.Heading>
+            <Text color="blue-200">{source.integrationLabel}</Text>
```
This is not a link anymore, so IMO it shouldn't have the color of a link in order not to confuse the user.

I propose a `teal-300` that will make it "stand out" as differentt, but not mark it as link
```suggestion
            <Text color="teal-300">{source.integrationLabel}</Text>
```

3nvi reviewed (approved):
Looks good @alexmylonas . Just a minor comment
